carbon-design-system/carbon
# fix(NumberInput): have less padding if number is valid

Closes #10155

Change padding for number input depending on whether icon is showing using the data-invalid property

#### Changelog

**Changed**

- padding for number input 


#### Testing / Reviewing

- Go to storybook playground for number inout
- set width for #root to 200px
- Set a max value to be very high
- Input number - Observe number goes right up to the edge of the decrement button
- Decrease the max value so that icon shows
- Observe number is clipped at the same level it is today


The new padding is right up to the decrement button when the icon is not showin, see below. This may not be desirable so let me know if the padding should be increased
<img width="659" alt="Screenshot 2023-03-31 at 21 54 43" src="https://user-images.githubusercontent.com/56260662/229229344-8f1361cb-b80a-4d40-b708-c4a8223bf81a.png">
